grammar.js

```diff
@@ -103,7 +103,10 @@ module.exports = grammar({
       optional(seq(
         $.php_tag,
         repeat($._statement)
-      ))
+      )),
+      // accept statements at top-level to enable
+      // parsing of partial code snippets (e.g, in markdown; see #168).
+      optional(repeat($._statement)),
```

Comment:
I'm not sure the statement will actually be parsed because of the optional($.text) at the top that may "eat" every tokens.

Reply:
Hmmm...you're right, I tried moving this line to the top of the `program` but the text still swallows everything up (see failing test)

Reply:
This is more complicated than I originally thought....do you have any idea on how to allow top-level statement parsing for injected code snippets while still allowing HTML?